### DISCUSSION TITLE
page keeps loading indefinitely

### DISCUSSION THREAD

**Discussion by JET-9** [dbarzin/deming#241] [Q&A]
Hello, I would like to test the solution. I installed it on an LXC container with Ubuntu 24.04 LTS. However, after following the documentation you provided, I started the server using php artisan serve --host 0.0.0.0 --port 8000. But when I try to access the site, the page keeps loading indefinitely.

**Comment by dbarzin** [comment]
Thank you for your interest in Deming and for taking the time to test it!  

The issue you’re describing might be related to one of the following points. Let’s troubleshoot step by step:  

1. **Check PHP Artisan Output**  
   When you start the server using `php artisan serve`, check the terminal output to ensure there are no errors or warnings.  

2. **Network Configuration**  
   Since you are running Deming in an LXC container, ensure that the container’s network settings allow connections to `0.0.0.0` on port `8000`. You might need to check or adjust port forwarding or firewall rules for the host and container.  

3. **Dependencies and Permissions**  
   - Confirm that all dependencies have been installed using `composer install`.  
   - Verify that the `.env` file is properly configured and that the application key has been generated using `php artisan key:generate`.  
   - Ensure correct folder permissions for `storage` and `bootstrap/cache` directories:  
     ```bash
     chmod -R 775 storage bootstrap/cache
     chown -R www-data:www-data storage bootstrap/cache
     ```  

4. **Accessing from the Host Machine**  
   Instead of `localhost`, try accessing the application using the IP address of the LXC container. For example:  
   ```
   http://<container-ip>:8000
   ```  

5. **Check the Laravel Logs**  
   Check the logs in `storage/logs/laravel.log` for any clues about the issue.  

6. **Consider a Web Server**  
   While `php artisan serve` is useful for development, it’s not recommended for production or testing in certain environments. You might want to configure a proper web server like Apache or Nginx with PHP-FPM for better compatibility.  

If none of the above steps resolve the issue, could you please provide additional information? For example:  
- Any specific error messages in the logs or browser console?  
- How is the LXC container's network configured (bridged, NAT, etc.)?  

Let me know how it goes, and I’ll be happy to help further!

**Comment by JET-9** [comment]
Thank you for your detailed response and the clarity of your troubleshooting steps. They were very helpful, and I greatly appreciate the speed and quality of your assistance!

After reviewing the setup, the issue was indeed related to user permissions. Your suggestion regarding folder permissions helped resolve it.

Additionally, I noticed a small point in the documentation: the command sudo a2dismod php8.3 results in the error:

Error: Module php8.3 does not exist
This can be resolved by running:

sudo apt install libapache2-mod-php8.3
This ensures the module is installed correctly and can be managed without issues.

Lastly, I have a question: is it possible to replace the MySQL database with PostgreSQL in Deming? If so, are there specific steps or considerations for making this change?

Once again, thank you for your excellent support and for taking the time to assist with these issues!

Best regards,

**Comment by dbarzin** [comment]
Thank you for your kind words! I'm glad to hear the troubleshooting steps were helpful.

Regarding your note about the documentation, you're absolutely right—thank you for pointing that out! I'll make sure to update the documentation to include the `sudo apt install libapache2-mod-php8.3` command as a prerequisite for enabling the module. It's great to have such precise feedback from users like you.

As for your question about replacing the MySQL database with PostgreSQL in Deming: Yes, it is possible to use PostgreSQL. Deming is built on Laravel, which supports multiple database systems, including PostgreSQL. Here are the key steps to make the switch:

 Modify the `.env` file to configure the connection details for PostgreSQL:
 ```plaintext
 DB_CONNECTION=pgsql
 DB_HOST=127.0.0.1
 DB_PORT=5432
 DB_DATABASE=your_database_name
 DB_USERNAME=your_username
 DB_PASSWORD=your_password
 ```

 If you're starting with a new PostgreSQL database, run the migrations to set up the schema:
 ```bash
 php artisan migrate --seed
 ```

**Comment by JET-9** [comment]
I’ve successfully set up the PostgreSQL database for Deming as per the instructions. However, I wanted to highlight one additional step that may be necessary for some users: installing the PostgreSQL driver for PHP. This can be done with the following command:
`sudo apt-get install php-pgsql`
Without this, Laravel may encounter issues connecting to the PostgreSQL database. I recommend including this in the documentation as a prerequisite for users switching to PostgreSQL.

I’m currently exploring and getting familiar with your tool. Once I’ve had the chance to fully understand its functionalities and use it more extensively, I’ll get back to you with detailed feedback to help improve it further.

**Comment by dbarzin** [comment]
Thank you for pointing this out and for your suggestion! You're absolutely right that the `php-pgsql` extension is crucial for Laravel to connect to PostgreSQL. The documentation will be updated to include it.

I really appreciate your efforts to explore Deming and look forward to your feedback. :+1:

**Comment by JET-9** [comment]
Hi,
I hope you're doing well. I wanted to let you know that I was testing the tool, and I encountered an error. To give you some context, I was on the "Action Plan" page and clicked on "New." That’s when the following error message appeared.

Interestingly, I don’t experience this issue when using the MySQL database. It seems to only happen with the PostgreSQL database.
Let me know if you need more details to investigate this!

Best regards,

![image](https://github.com/user-attachments/assets/8492b84f-6430-4643-8c71-40df732b4bed)
![image](https://github.com/user-attachments/assets/3e0042d5-d76d-42a1-8262-cb0946cee796)
![image](https://github.com/user-attachments/assets/2b90d82c-ae10-4284-b49d-e951f2b43176)
![image](https://github.com/user-attachments/assets/da1c2116-fbbe-466b-b227-0bbb23511f44)

**Comment by dbarzin** [comment]
An issue has been created : https://github.com/dbarzin/deming/issues/243

**Comment by dbarzin** [comment]
Fixed here : https://github.com/dbarzin/deming/pull/244/commits/0e1ba3ab0000821928bc3eae822c3462fd4b76e7
